### DISCUSSION TITLE
Flux2 fails with Git Over SSH

### DISCUSSION THREAD

**Discussion by amralkar** [fluxcd/flux2#4592] [General]
I configured git server on Ubuntu machine and I use it in my projects. I was trying to set the FluxCD2 for my EKS cluster but I get error when I create git source

```
❯ flux create source git flux-system \
 --url=ssh://ubuntu@git.mydomain.pvt:abhishek/demo-eks.git \
 --branch=main \
 --secret-ref=flux-system \
 --interval=5m \
 --export > flux2-sync.yaml
```

I get below error

``` git URL parse failed: parse "ssh://ubuntu@git.mydomain.pvt:abhishek/demo-eks.git": invalid port ":abhishek" after host```

I cannot use

```
ssh://ubuntu@git.mydomain.pvt/abhishek/demo-eks.git
```

**Comment by somtochiama** [comment]
please see: https://fluxcd.io/flux/components/source/gitrepositories/#url
The right url should be ssh://ubuntu@git.mydomain.pvt/abhishek/demo-eks.git

**Reply by amralkar** [reply]
In git over SSH when I use that URL it says no repo found.

```
git clone ubuntu@git.mydomain.pvt/abhishek/demo-eks.git
fatal: repository 'ubuntu@git.mydomain.pvt/abhishek/demo-eks.git' does not exist```

Same error Flux gives.

**Reply by geotransformer** [reply]
flux 2.2.3. Same observation

flux-deploy-poc-ssh ssh://[git@github.com](mailto:git@github.com):xxxx/flux-deploy-poc.git 10m False failed to parse url 'ssh://[git@github.com](mailto:git@github.com):xxxxx/flux-deploy-poc.git': parse "ssh://[git@github.com](mailto:git@github.com):xxxxx/cndp-deploy-poc.git": invalid port ":xxxxx" after host

**Reply by pietro** [reply]
Flux behaviour is documented but what's the rationale behind it? The library it uses for git interactions, go-git, supports the scp style URLs.

**Comment by oismael-arabwork** [comment]
issue still occurs
